api-2.2/src/main/java/org/openmrs/module/fhir2/api/translators/impl/MedicationRequestStatusTranslatorImpl_2_2.java

```diff
@@ -0,0 +1,35 @@
+package org.openmrs.module.fhir2.api.translators.impl;
+
+import javax.annotation.Nonnull;
+
+import org.hl7.fhir.r4.model.MedicationRequest;
+import org.openmrs.DrugOrder;
+import org.openmrs.Order;
+import org.openmrs.annotation.OpenmrsProfile;
+import org.openmrs.module.fhir2.api.translators.MedicationRequestStatusTranslator;
+import org.springframework.stereotype.Component;
+
+@Component
+@OpenmrsProfile(openmrsPlatformVersion = "2.2.* - 2.*")
+public class MedicationRequestStatusTranslatorImpl_2_2 implements MedicationRequestStatusTranslator {
+	
+	@Override
+	public MedicationRequest.MedicationRequestStatus toFhirResource(@Nonnull DrugOrder drugOrder) {
+		if (drugOrder == null) {
+			return null;
+		}
+		
+		if (drugOrder.getFulfillerStatus() != null
+		        && drugOrder.getFulfillerStatus().equals(Order.FulfillerStatus.COMPLETED)) {
+			return MedicationRequest.MedicationRequestStatus.COMPLETED;
+		} else if (drugOrder.isActive()) {
+			return MedicationRequest.MedicationRequestStatus.ACTIVE;
+		} else if (drugOrder.isDiscontinuedRightNow() || drugOrder.getVoided()) {
+			return MedicationRequest.MedicationRequestStatus.CANCELLED;
```

Comment:
I'm not sure I'd equate discontinued = canceled and expired = stopped.  Is this logic confirmed from, say, @bmamlin ?

Reply:
FWIW, the discontinued and expired logic isn't new to this commit, and already exists in the main translator (https://github.com/openmrs/openmrs-module-fhir2/blob/master/api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/MedicationRequestStatusTranslatorImpl.java)  (I considered having one inherit from the other, but it didn't seem worth the effort.)

That being said, this logic *is* relatively new and could still be up for debate You can see the implemented logic and discussion here:

https://issues.openmrs.org/browse/FM2-480

FWIW, I like cancelled for discontinued, but am not in love with "stopped" for "expired", but there didn't seem to be a better option, at least until we have "ended" in FHIR 5.